core/lib/Cardano/Address/Errors.hs

```diff
@@ -0,0 +1,61 @@
+{-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE StandaloneDeriving #-}
+
+-- |
+-- Copyright: © 2020 IOHK
+-- License: Apache-2.0
+
+module Cardano.Address.Errors
+    ( -- $overview
+
+      -- * AddrError
+      ShelleyAddrError(..)
+    , ByronAddrError(..)
+    , IcarusAddrError(..)
+    , JormungandrAddrError(..)
+  ) where
```

Comment:
Wouldn't it be better to have errors specific to each modules inside the module they relate to (or close)? These errors share nothing in common apart from the fact that they are errors. If there was a common and abstract "Error" data-structure that had some well-defined properties / interfaces that would indeed deserve a separate module. Here however, it seems to me that errors are moved in a separate module for no apparent reasons :thinking: ?

Reply:
I have no strong opinion on that. However, data type modules are often used to avoid circular imports. I naturally prefer to keep data out of function modules. Seeing all at once also enables me to easily see what structure is shared.